### DISCUSSION TITLE
Database and IO strategy

### DISCUSSION THREAD

**Discussion by michaelosthege** [Akkudoktor-EOS/EOS#108] [Ideas]
At the moment EOS uses one table in a MariaDB and makes three different queries to it.
This table is also the point of data exchange.

Just because the prototype was built with a relational database doesn't mean that EOS should stick to that and use an ORM.
The app is a Flask server, so data exchange could also be done with GET/PUT endpoints, without the need for a database.

I think this is really a question of "How is data provided to EOS, and how does EOS provide data to other home automation components?".

As of 2024-10-06 the code is tailored to Andreas' deployment, and I think it's important that he continues to have great integration with his setup.

**Comment by drbacke** [comment]
No DB should be necessary for the Flask_server. 
The reason for the DB is only the local SoC function, which I can gladly remove it.

**Reply by drbacke** [reply]
Completed in #112

**Comment by BerndCzech** [comment]
I guess this app will probably some kind of state. 

Should we start with some **sqlite** when we need it?

**Reply by drbacke** [reply]
I already removed All DB Connections.
There is No need for
